lntest/itest/lnd_multi-hop_htlc_aggregation_test.go

```diff
@@ -306,81 +268,53 @@ func testMultiHopHtlcAggregation(net *lntest.NetworkHarness, t *harnessTest,
 	// levels to be aggregated into one tx. For earlier channel types, they
 	// will be separate transactions.
 	if hasAnchors {
-		require.Len(t.t, timeoutTxs, 1)
-		require.Len(t.t, successTxs, 1)
+		require.Len(ht, timeoutTxs, 1)
+		require.Len(ht, successTxs, 1)
 	} else {
-		require.Len(t.t, timeoutTxs, numInvoices)
-		require.Len(t.t, successTxs, numInvoices)
+		require.Len(ht, timeoutTxs, numInvoices)
+		require.Len(ht, successTxs, numInvoices)
 	}
 
 	// All mempool transactions should be spending from the commitment
 	// transaction.
-	assertAllTxesSpendFrom(t, txes, closeTxid)
+	ht.AssertAllTxesSpendFrom(txes, closeTxid)
 
-	// Mine a block to confirm the transactions.
-	block := mineBlocks(t, net, 1, expectedTxes)[0]
-	require.Len(t.t, block.Transactions, expectedTxes+1)
+	// Mine a block to confirm the all the transactions, including Carol's
+	// commitment tx, anchor tx(optional), and the second-level timeout and
+	// success txes.
+	block := ht.Miner.MineBlocksAndAssertNumTxes(1, expectedTxes)[0]
+	require.Len(ht, block.Transactions, expectedTxes+1)
 
 	// At this point, Bob should have broadcast his second layer success
 	// transaction, and should have sent it to the nursery for incubation,
 	// or to the sweeper for sweeping.
-	err = waitForNumChannelPendingForceClose(
-		bob, 1, func(c *lnrpcForceCloseChannel) error {
-			if c.Channel.LocalBalance != 0 {
-				return nil
-			}
+	ht.AssertNumPendingForceClose(bob, 1)
 
-			if len(c.PendingHtlcs) != 1 {
-				return fmt.Errorf("bob should have pending " +
-					"htlc but doesn't")
-			}
-
-			if c.PendingHtlcs[0].Stage != 1 {
-				return fmt.Errorf("bob's htlc should have "+
-					"advanced to the first stage but was "+
-					"stage: %v", c.PendingHtlcs[0].Stage)
-			}
-
-			return nil
-		},
-	)
-	require.NoError(t.t, err)
+	// For this channel, we also check the number of HTLCs and the stage
+	// are correct.
+	ht.AssertNumHTLCsAndStage(bob, bobChanPoint, numInvoices*2, 2)
```

Comment:
In the old check, because we exit at this line `if c.Channel.LocalBalance != 0 {`, this pending channel's HTLCs are never checked. And if I got it right, I think the stage is set to 2 here,
https://github.com/lightningnetwork/lnd/blob/b387e2c718e70afdbf8dff69c02aa7a706ca7dec/contractcourt/htlc_timeout_resolver.go#L439-L442